src/panels/lovelace/cards/hui-iframe-card.ts

```diff
@@ -0,0 +1,86 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import "../../../components/ha-card.js";
+
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+interface Config extends LovelaceConfig {
+  aspect_ratio?: string;
+  title?: string;
+  url: string;
+}
+
+export class HuiIframeCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected config?: Config;
+
+  static get properties() {
+    return {
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 1 + this.offsetHeight / 50;
+  }
+
+  public setConfig(config: Config) {
+    if (!config.url) {
+      throw new Error("URL required");
+    }
+
+    this.config = config;
+    this.requestUpdate();
+  }
+
+  protected render() {
+    if (!this.config) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this.config.title}">
+        <div id="root">
+          <iframe src="${this.config.url}"></iframe>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  private renderStyle() {
+    if (!this.config) {
```

Comment:
No need for this check because we only call this in `render()` after the check. Instead, inside the CSS make it `this.config!.aspect_ratio`. The `!` will tell TS that we know it's not null.

Reply:
TIL. Thanks!